guide/english/abap/index.md

```diff
@@ -0,0 +1,125 @@
+
+
+METHOD filter_data_column.
+
+    " Apply some constant to a variable, that is one of multiple ways of doing that dynamically, "lc_constant"'s type is created and fitted dynamically
+    DATA(lc_constant) = 'If this string will be found, method will be executed. Otherwise check will exit the class.'.
+
+    " Check if this method call provided an argument for this class to be executed 
+    " Check if text value provided by the parameter contains pattern stored in previously created variable  
+    CHECK pv_some_desc CP |*{ lc_constant }*| .
+
+
+      " Use OPEN-SQL to look up data from system or user defined global tables 
+      " internal variables and dynamically created tables are exited with @ / @DATA
+      SELECT DISTINCT something FROM some_table
+          WHERE counter = ( SELECT MIN( counter ) FROM
+          some_table WHERE some_column = @pv_sourcesys )
+        AND soursystem = @pv_sourcesys
+         INTO @DATA(lv_yft_gen_repr_sel_pck).
+      ENDSELECT.
+
+    " Check if proper value was selected - if so, no deletion should happen
+    CHECK lv_yft_gen_repr_sel_pck IS NOT INITIAL AND lv_yft_gen_repr_sel_pck NE '00'.
+
+    " Loop at internal table given to the method assigning pointers - field symbols, apply filtering to it - if matches, than delete
+    LOOP AT result_package ASSIGNING FIELD-SYMBOL(<fs_result_package>).
+      "As this table has type 'any' - indexed table, than we have to assign specific column name of the structure to a field symbol
+      ASSIGN COMPONENT 'NAME_OF_THE_COLUMN_WE_ARE_LOOKING_FOR' OF STRUCTURE <fs_result_package> TO FIELD-SYMBOL(<fs_NAME_OF_THE_COLUMN_WE_ARE_LOOKING_FOR>).
+      CHECK <fs_gm1pck> IS ASSIGNED.
+      IF <fs_gm1pck> NE lv_yft_gen_repr_sel_pck.
+        DELETE result_package.
+      ENDIF.
+    ENDLOOP.
+
+
+ENDMETHOD. 
+
+```
+
+## ABAP Execution Layers
+* Presentation layer 
+This layer is ditributed to the end users and can be accessed both from WEB and from the SAP software itself. In ERP - ECC systems, this layer is when the user creates an invoive, order, executes reports.
+
+* Application layer 
+This layer on application servers. It constists of ABAP runtime environment in which ABAP programs are executed.
+
+* Database layer
+Consists of a database system in which the central dataset of an AS ABAP is saved.
+
+## ABAP execution scenarios
+
+ABAP programms can transfer data for processiong from database layer to application layer making it a bottom-up approach. This approach makes an application to run a bit slower because it has to transfer the data between the servers.
+To speed up the processing, a top-town approach was introduced. It allows the user to write code that is going to send the instruction (procedure) to the database of how the data should be changed without having to fetch it into application layer. The code is still written in the application layer as a ABAP Managed Database Procedure usually of type SQL Script.  
+
+
```

Comment:
Everything below this can probably be deleted if there isn't going to be any information added.